### PR TITLE
Document why Pixie uses a different loop condition than the official BCC

### PR DESCRIPTION
**Pull request by yzhao1012**

### CODE REVIEW

**Comment by yzhao1012** [comment]
@JamesMBartlett @oazizi000

File: src/cc/perf_reader.c [hunk 1]
```diff
@@ -172,6 +172,21 @@ void perf_reader_event_read(struct perf_reader *reader) {
...
+  // 
+  // NOTE: The loop condition is different from the official BCC repo. The official impl code is
+  // as follows:
+  // for (data_head = read_data_head(perf_header); perf_header->data_tail != data_head;
+  //      data_head = read_data_head(perf_header)) {
+  //     ... Loop body
+  // }
+  //
+  // As stated in https://github.com/iovisor/bcc/pull/3822#issuecomment-1023775038,
+  // the official BCC code above can handle occasional burst by eagerly depleting the perf buffer.
```
Comment (oazizi000) [inline src/cc/perf_reader.c]: ...can handle occasional bursts...

File: src/cc/perf_reader.c [hunk 2]
```diff
@@ -172,6 +172,21 @@ void perf_reader_event_read(struct perf_reader *reader) {
...
+  //
+  // The worst case scenario is that the producer constantly produces more data
+  // than the consumer can keep up, and results into an infinite loop. Pixie team have observed
+  // a few cases where an perf buffer polling iteration took very long time, which was fixed
+  // by this new loop condition.
```
Comment (oazizi000) [inline src/cc/perf_reader.c]: ...than the consumer can process, and results in an infinite loop. The Pixie team has observed
a few cases where a perf buffer polling iteration took a very long time, which was fixed
by this new loop condition.

**Review by oazizi000 — COMMENTED** [review]
Cool. Some grammatical tweaks.

**Review by oazizi000 — APPROVED** [review]

**Comment by oazizi000** [comment]
Merged in f1a624f67700efd29f7ea7706e5f1754845935b5

**Pull request closed without merging**